src/config.c

```diff
@@ -20,28 +20,107 @@
 #include "resp3.h"
 #include "util/workers.h"
 
-#include "util/config_macros.h"
-
 #define __STRINGIFY(x) #x
 #define STRINGIFY(x) __STRINGIFY(x)
 
 #define RS_MAX_CONFIG_TRIGGERS 1 // Increase this if you need more triggers
 RSConfigExternalTrigger RSGlobalConfigTriggers[RS_MAX_CONFIG_TRIGGERS];
 
+int set_numeric_config(const char *name, long long val, void *privdata,
+                  RedisModuleString **err) {
+  REDISMODULE_NOT_USED(name);
+  REDISMODULE_NOT_USED(err);
+  *(long long *)privdata = val;
+  return REDISMODULE_OK;
+}
+
+long long get_numeric_config(const char *name, void *privdata) {
+  REDISMODULE_NOT_USED(name);
+  return (*(long long *)privdata);
+}
+
+int set_uint_numeric_config(const char *name, long long val,
+                           void *privdata, RedisModuleString **err) {
+  REDISMODULE_NOT_USED(name);
+  REDISMODULE_NOT_USED(err);
+  *(unsigned int *)privdata = (unsigned int) val;
+  return REDISMODULE_OK;
+}
+
+long long get_uint_numeric_config(const char *name, void *privdata) {
+  REDISMODULE_NOT_USED(name);
+  return (long long)(*(unsigned int *)privdata);
+}
+
+int set_bool_config(const char *name, int val, void *privdata,
+                    RedisModuleString **err) {
+  REDISMODULE_NOT_USED(name);
+  REDISMODULE_NOT_USED(err);
+  *(int *)privdata = val;
+  return REDISMODULE_OK;
+}
+
+int set_inverted_bool_config(const char *name, int val, void *privdata,
+                             RedisModuleString **err) {
+  REDISMODULE_NOT_USED(name);
+  REDISMODULE_NOT_USED(err);
+  *(int *)privdata = (val == 0);
+  return REDISMODULE_OK;
+}
+
+int get_bool_config(const char *name, void *privdata) {
+  REDISMODULE_NOT_USED(name);
+  return *(int *)privdata;
+}
+
+int get_inverted_bool_config(const char *name, void *privdata) {
+  REDISMODULE_NOT_USED(name);
+  return !*(int *)privdata;
+}
+
+int set_immutable_string_config(const char *name, RedisModuleString *val, void *privdata,
+                      RedisModuleString **err) {
+  REDISMODULE_NOT_USED(name);
+  REDISMODULE_NOT_USED(err);
+  char **ptr = (char **)privdata;
+  if (val) {
+    size_t len;
+    const char *ret = RedisModule_StringPtrLen(val, &len);
+    if (len > 0) {
+      *ptr = rm_strndup(ret, len);
+    }
+  }
+  return REDISMODULE_OK;
+}
+
 // EXTLOAD
 CONFIG_SETTER(setExtLoad) {
   int acrc = AC_GetString(ac, &config->extLoad, NULL, 0);
   RETURN_STATUS(acrc);
 }
 
 CONFIG_GETTER(getExtLoad) {
-  if (config->extLoad) {
+  if (config->extLoad && strlen(config->extLoad) > 0) {
```

Comment:
maybe you want to use sdslen instead?

Reply:
Isn't `extLoad` a plain char*, and not an sds string?